core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/util/DatabaseUtil.java

```diff
@@ -347,28 +366,27 @@ public static int getIntegerValueFromDatabase(Connection dbConnection, String sq
                     }
                 }
             }
-            rs = prepStmt.executeQuery();
-            if (rs.next()) {
-                value = rs.getInt(1);
+            try(ResultSet rs = prepStmt.executeQuery()) {
+                if (rs.next()) {
+                    value = rs.getInt(1);
+                }
+                return value;
             }
-            return value;
         } catch (SQLException e) {
             String errorMessage = "Using sql : " + sqlStmt + " " + e.getMessage();
             if (log.isDebugEnabled()) {
                 log.debug(errorMessage, e);
             }
             throw new UserStoreException(errorMessage, e);
-        } finally {
-            DatabaseUtil.closeAllConnections(null, rs, prepStmt);
+        }finally {
+            close(dbConnection);
```

Comment:
Since the connection getting passed as a parameter to this method, it shouldn't close the connection inside this method.

Need to close the connection in the original place where it originally (create and )close the dbConnection.